java/arcs/core/analysis/InformationFlow.kt

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.Check
+import arcs.core.data.Claim
+import arcs.core.data.HandleConnectionSpec
+import arcs.core.data.HandleMode
+import arcs.core.data.InformationFlowLabel
+import arcs.core.data.InformationFlowLabel.Predicate
+import arcs.core.data.InformationFlowLabel.SemanticTag
+import arcs.core.data.Recipe
+import arcs.core.data.Recipe.Particle
+import arcs.core.util.TaggedLog
+import java.util.BitSet
+
+/**
+ * Information flow analysis of a [Recipe] using the labels in the particle checks and claims.
+ *
+ * For the time being, we pass in the ingress handles.
```

Comment:
Might be worth adding a TODO comment here

Reply:
Done.